crates/host/src/wasmbus/mod.rs

```diff
@@ -3210,38 +3352,69 @@ impl Host {
         Ok((config, secrets))
     }
 
+    /// Validates that the provided configuration names exist in the store and are valid.
+    ///
+    /// For any configuration that starts with `SECRET_`, the configuration is expected to be a secret reference.
+    /// For any other configuration, the configuration is expected to be a [`HashMap<String, String>`].
+    async fn validate_config(&self, config_names: Vec<String>) -> anyhow::Result<()> {
+        let config_store = self.config_data.clone();
+        let validation_errors = futures::future::join_all(
+            config_names.iter().map(|config_name| {
+                let config_store = config_store.clone();
+                async move {
+                    match config_store.get(config_name).await {
+                        Ok(Some(secret)) if config_name.starts_with(SECRET_PREFIX) => serde_json::from_slice::<crate::secrets::SecretReference>(&secret)
+                            .map(|_| ())
+                            .map_err(|_| anyhow::anyhow!("failed to deserialize secret reference from config store, ensure {config_name} is a secret reference and not configuration")),
+                        Ok(Some(config)) => serde_json::from_slice::<HashMap<String, String>>(&config)
+                            .map(|_| ())
+                            .map_err(|_| anyhow::anyhow!("failed to deserialize config from config store, ensure {config_name} is a valid string -> string map")),
+                        Ok(None) => bail!("Configuration {config_name} not found in config store"),
+                        Err(e) => bail!(e),
+                    }
+                }
+            }),
+        )
+        .await;
+
+        let errors = validation_errors
+            .into_iter()
+            .filter_map(|res| res.err())
+            .collect::<Vec<_>>();
+        if !errors.is_empty() {
+            let mut error_chain = anyhow::anyhow!("failed to validate link config");
+            for error in errors {
+                error_chain = error_chain.context(error);
```

Comment:
I think this ends up adding an error chain which is not quite what this means. Each link config validation didn't cause the next thing in the chain. I would maybe return an error that is a `(Error, &str)` so you can construct the name of each config that failed and the error that occurred. Then join that into a big error

Reply:
How would you join into a big error here? I can refactor fairly easily to simplify this chain, but I don't see a way to combine those errors

Reply:
Was just thinking to join all the errors into a larger string message to add as context